packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -532,25 +666,40 @@ class _RenderableTileLayer extends _RenderableLayer<TileLayer> {
         //  * stagger: Hexagonal tiles move down or right by only a fraction,
         //             specifically 3/4 the width or height, for packing.
         if (_map.staggerAxis == StaggerAxis.y) {
-          offsetX = tx * _map.tileWidth + staggerX + halfDestinationTile.x;
+          offsetX = tx * size.x + staggerX + halfDestinationTile.x;
           offsetY = ty * staggerY + halfDestinationTile.y;
         } else {
           offsetX = tx * staggerX + halfDestinationTile.x;
-          offsetY = ty * _map.tileHeight + staggerY + halfDestinationTile.y;
+          offsetY = ty * size.y + staggerY + halfDestinationTile.y;
         }
 
         final scos = flips.cos * scale;
         final ssin = flips.sin * scale;
 
-        batch.addTransform(
-          source: src,
-          transform: ui.RSTransform(
-            scos,
-            ssin,
-            offsetX + -scos * anchorX + ssin * anchorY,
-            offsetY + -ssin * anchorX - scos * anchorY,
-          ),
-          flip: flips.flip,
+        final transform = ui.RSTransform(
+          scos,
+          ssin,
+          offsetX + -scos * anchorX + ssin * anchorY,
+          offsetY + -ssin * anchorX - scos * anchorY,
+        );
+
+        // A second pass is only needed in the case of staggery.
+        if (_map.staggerAxis == StaggerAxis.x && staggerY > 0) {
+          xSecondPass.add(_Transform(src, transform, flips.flip, batch));
```

Comment:
It will add some overhead with creating all these new objects, but maybe there is no good way around it.

Reply:
You either do this, or you do:

- for (even){}, for(odd){}. // stagger x even 
- for (odd){}, for(even){}. // stagger x odd

 What was told me by some of the Dart VM Guys: create a lot of garbage quickly and give it up - new heap is fast.
 
 Honestly, the transforms on SpriteBatch should be fungible. I should be able to reorder and update individual elements.

Reply:
When did they say this? Because we've had a lot of problems with creating too many objects previously.

Reply:
It was certainly a while ago! That said, this is only done on cache refresh - not every frame, so this is totally fine I think.

Reply:
Yeah, we can just re-visit this if we notice any performance issues here.